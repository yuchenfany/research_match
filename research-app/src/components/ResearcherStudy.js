/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import '../assets/index.css';
import NavBar from './NavBar';
import { getStudyById } from '../modules/study-api';

// TO DO: add back in studyId prop
function ResearcherStudy({
  study, setStudy, user,
}) {
  const navigate = useNavigate();

  useEffect(() => {
    getStudyById(study.studyId)
      .then(setStudy);
  }, []);

  return (
    <div className="Study Page">
      <NavBar user={user} />
      <div className="study-flex">
        <div className="header-left">
          {study.title}
        </div>
        <div>
          Duration:
          {study.duration}
        </div>
        <div>
          Compensation:
          {study.compensation}
        </div>
        <div>
          Researcher names: &nbsp;
          {study?.researchers}
        </div>
        <button className="button" type="button" onClick={() => navigate('/edit-study')}>Edit Study</button>
        <div className="header-small"> Description </div>
        <div className="paragraph">
          {study.description}
        </div>
      </div>
    </div>
  );
}

export default ResearcherStudy;
