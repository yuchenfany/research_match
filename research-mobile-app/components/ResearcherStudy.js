/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */

import React, { useState, useEffect } from 'react';
import NavBar from './NavBar';

function ResearcherStudy({ route, navigation}) { // add props user
  let {user, setUser, study, setStudy } = route.params;
  const [enrolledStudies, setEnrolledStudies] = useState([]);
  // const navigate = useNavigate();
  async function getStudy() {
    console.log(study.studyId);
    const studyData = await fetch(`http://localhost:5000/study/${study.studyId}`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    const data = await studyData.json();
    console.log(data);

    return data;
  }

  useEffect(() => {
    getStudy()
      .then(setStudy);
  }, []);

  const goToEditStudy = () => {
    navigation.navigate('EditStudy', {
      user,
      setUser,
      study,
      setStudy,
    });
  }

  return (
    <View className="Study Page">
      <NavBar user={user} setUser={setUser} navigation={navigation} />
      <View className="study-flex">
        <Text className="header-left">
          {study.title}
        </Text>
        <Text>
          Duration:
          {study.duration}
        </Text>
        <Text>
          Compensation:
          {study.compensation}
        </Text>
        <Text> Researcher names: [this is the researcherStudy page] </Text>
        <Button className="button" onClick={() => goToEditStudy()}>Edit Study</Button>
        <Text className="header-small"> Description </Text>
        <Text className="paragraph">
          {study.description}
        </Text>
      </View>
    </View>
  );
}


export default ResearcherStudy;
