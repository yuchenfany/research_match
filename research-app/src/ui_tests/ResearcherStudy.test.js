/**
 * @jest-environment jsdom
 */

/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
// import { jest } from '@jest/globals';

import ResearcherStudy from '../components/ResearcherStudy';

// const mockedUsedNavigate = jest.fn();

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useNavigate: () => mockedUsedNavigate,
// }));

test('Title label', () => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ResearcherStudy study={study} setStudy={mockSetStudy} user={user1} setUser={mockSetUser} status={status} setStatus={mockSetStatus} />
        </Router>);  
  const label = screen.getByText("Description");
  expect(label).toBeInTheDocument();
});

// test('participant clicked',() => {
//   render(
//     <Router>
//         <Study />
//     </Router>); 
// // fireEvent.click(screen.getByTestId('participant'));
// expect(screen.getByTestId('participant')).toBeInTheDocument();
// //   expect(screen.getByTestId('participant')).toBeChecked();
// });

test('Enroll Button',() => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ResearcherStudy study={study} setStudy={mockSetStudy} user={user1} setUser={mockSetUser} status={status} setStatus={mockSetStatus} />
        </Router>);  
  const linkElement = screen.getByRole('button', {
    name: /Edit Study/i,
  });
  fireEvent.click(linkElement);
  expect(linkElement).toBeInTheDocument();
});

// // test('Researcher Button',() => {
// //   const mockSetUser = jest.fn();
// //   const user1 = { username: 'user1', password: 'user1pass' };
// //   render(
// //     <Router>
// //       <Type setUser={mockSetUser} user={user1} />
// //     </Router>,
// //   );
// //   const linkElement = screen.getByRole('radio', {
// //     name: /researcher/i
// //   });
// //   fireEvent.click(linkElement);
// //   // expect(mockSetUser).toBeCalled();
// //   expect(linkElement).toBeInTheDocument();
// // });
// test('researcher clicked',() => {
//   render(
//     <Router>
//         <Study />
//     </Router>); 
//   // fireEvent.click(screen.getByTestId('researcher'));
//   expect(screen.getByTestId('researcher')).toBeInTheDocument();
//   // expect(screen.getByTestId('researcher')).toBeChecked();
// });

test('snapshot test', () => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
  const component = renderer.create( 
    <Router>
        <ResearcherStudy study={study} setStudy={mockSetStudy} user={user1} setUser={mockSetUser} status={status} setStatus={mockSetStatus} />
    </Router>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});