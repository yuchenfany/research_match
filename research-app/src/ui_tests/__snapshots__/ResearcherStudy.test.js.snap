// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="Study Page"
>
  <div
    className="nav"
  >
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      HOME
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      MY STUDIES
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      MESSAGES
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      ANALYTICS
    </button>
    <button
      className="account-btn"
      onClick={[Function]}
      type="button"
    >
      ACCOUNT
    </button>
    <div
      className="acct-options"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <button
        className="account-btn"
        onClick={[Function]}
        type="button"
      >
        EDIT PROFILE
      </button>
      <button
        className="account-btn"
        onClick={[Function]}
        type="button"
      >
        DELETE ACCOUNT
      </button>
      <button
        className="logout-btn"
        onClick={[Function]}
        type="button"
      >
        LOGOUT
      </button>
    </div>
  </div>
  <div
    className="study-flex"
  >
    <div
      className="header-left"
    />
    <div>
      Duration:
    </div>
    <div>
      Compensation:
    </div>
    <div>
       Researcher names: [this is the researcherStudy page] 
    </div>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Edit Study
    </button>
    <div
      className="header-small"
    >
       Description 
    </div>
    <div
      className="paragraph"
    />
  </div>
</div>
`;
